org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/utils/XMLGenerator.java

```diff
@@ -149,6 +197,8 @@ private int generate(CMElementDeclaration elementDeclaration, String prefix, boo
 		if (generatedElements.contains(elementDeclaration)
 				|| existingElementNames.contains(elementDeclaration.getLocalName())) {
 			return snippetIndex;
+		}else if (level > 5){
+			return snippetIndex;

```

Comment:
Could you give me the full sample which causes problem please to check level?

Reply:
If I use https://github.com/eclipse/lemminx/blob/main/org.eclipse.lemminx/src/test/resources/relaxng/tei_all.rng

Reply:
Could you please write a test with this usecase

Reply:
I have added a test for this case and it seems to be working, but returns a lot to content. I wonder if it's a good idea to restrict the level of children we generate?